app/models/user.rb

```diff
@@ -47,6 +47,10 @@ class User < ApplicationRecord
 
   # `Other` is a catch-all for any status that doesn't fit into the other categories and should be *last*.
   enum :status_group, {unknown: 0, learner: 2, educator: 3, other: 1}, default: :unknown, prefix: true
+  # When a user is created through the NBP wallet connection, we want to ensure a valid status group.
+  validates :status_group, inclusion: {in: %w[learner educator], message: :unrecognized_role}, on: :create, if: lambda {
```

Comment:
`Relationship#userdata[:status_group]` returns the two valid values as symbols.
```suggestion
  validates :status_group, inclusion: {in: %s[learner educator], message: :unrecognized_role}, on: :create, if: lambda {
```

Reply:
The correct syntax would be

```suggestion
  validates :status_group, inclusion: {in: %s[learner educator], message: :unrecognized_role}, on: :create, if: lambda {
```

However, an array of symbols doesn't work, since a symbol or integer passed is automatically converted. I extended the specs and kept the original behavior.